### ISSUE TITLE
aria-posinset becomes 0 once community-list node is opened

### ISSUE THREAD

**Issue by AAwouters** [DSpace/dspace-angular#4068]
## Describe the bug
Angular/cdk version 18 adds the `aria-posinset` attribute to `cdk-tree-node`s.
This attribute should have a value starting from 1 indicating what the position of the node is in the set.
When opening the node to show the sub-communities, this value becomes 0 which is not a valid value.

## To Reproduce
1. Navigate to the `/community-list` page ('Communities & Collections' button in the navbar)
2. Inspect the community nodes using browser tools & look for the `aria-posinset` property on the `cdk-tree-node`s
3. Click on the node
4. Notice that the `aria-posinset` value becomes 0

## Expected behavior
The `cdk-tree-node` should retain its original `aria-posinset` value.

## Related work
Introduced with #3717 

## Actions to take once resolved
Once this issue is resolved, the `aria-valid-attr-value` test in the [community-list e2e test](https://github.com/DSpace/dspace-angular/blob/c90b05650aa597adfb7592c10845c48ccd55c51a/cypress/e2e/community-list.cy.ts#L16-L21) should be re-enabled.

## Additional information
From the angular/cdk source code:
The `aria-posinset` value is determined using `findIndex` in the [_getPositionInSet method](https://github.com/angular/components/blob/main/src/cdk/tree/tree.ts#L863). 
`findIndex` returns -1 when no matching entry can be found, so this might explain where the 0 value comes from.
Why exactly no matching entry can be found will require deeper investigation.
